src/vs/platform/native/electron-main/nativeHostMainService.ts

```diff
@@ -848,11 +848,65 @@ export class NativeHostMainService extends Disposable implements INativeHostMain
 	}
 
 	private getTargetWindow(fallbackWindowId: number | undefined): ICodeWindow | IAuxiliaryWindow | undefined {
-		let window = this.instantiationService.invokeFunction(getFocusedOrLastActiveWindow);
-		if (!window) {
-			window = this.windowById(fallbackWindowId);
-		}
+		return this.instantiationService.invokeFunction(getFocusedOrLastActiveConnectedWindow, fallbackWindowId);
+	}
+}
+
+function isConnectedWindow(window: ICodeWindow | IAuxiliaryWindow | undefined, other: ICodeWindow | IAuxiliaryWindow | undefined): boolean {
+	if (!window || !other) {
+		return false;
+	}
+
+	if (window.id === other.id) {
+		return true;
+	}
+
+	if (isIAuxiliaryWindow(window) && window.parentId === other.id || isIAuxiliaryWindow(other) && other.parentId === window.id) {
+		return true;
+	}
+
+	// we could inspect all the parents on both sides but afaik the nesting only goes one deep?
+
+	return false;
+}
 
-		return window;
+function getFocusedOrLastActiveConnectedWindow(accessor: ServicesAccessor, windowId: number | undefined): ICodeWindow | IAuxiliaryWindow | undefined {
```

Comment:
i had to move this out of `src/vs/platform/windows/electron-main/windows.ts` to avoid a cyclical dependency